src/theory/strings/base_solver.cpp

```diff
@@ -374,16 +375,31 @@ bool BaseSolver::processConstantLike(Node a, Node b)
         // s or t may be concrete integers corresponding to code
         // points of string constants, and thus are not guaranteed to
         // be terms in the equality engine.
-        d_im.sendInference(exp, exp, conc, InferenceId::STRINGS_UNIT_INJ_OOB);
+        NodeManager* nm = NodeManager::currentNM();
+        // We must send this lemma immediately, since otherwise if buffered,
+        // this lemma may be dropped if there is a fact or conflict that
+        // preempts it.
```

Comment:
Perhaps an SMT-LIB (strings) run is appropriate here before merging? This might have performance effect, right?

Reply:
No, this code is only run with `--strings-code-elim`, which is expert and disabled by default.